app/controllers/my/tracks_controller.rb

```diff
@@ -43,7 +43,23 @@ def show
     else
       core_exercises, side_exercises = normal_exercises.partition {|e|e.core?}
 
-      @core_exercises_and_solutions = core_exercises.map{|e|[e, mapped_solutions[e.id]]}
+      core_exercises = core_exercises.map do |e|
+        ExerciseWithSolution.new(e, mapped_solutions[e.id])
+      end
+
+      @core_exercises_and_solutions = core_exercises.
+        inject([[], [], []]) do |collection, exercise|
+          if exercise.completed?
+            collection[0] << exercise
+          elsif exercise.unlocked?
+            collection[1] << exercise
+          elsif exercise.locked?
+            collection[2] << exercise
+          end
+
+          collection
+        end.
```

Comment:
And maybe do this as `{}` rather than `do end` to make the `end.` a little nicer?